src/components/pagination/index.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { FC, PropsWithChildren, useMemo } from 'react';
+import { useMultiStyleConfig } from '@chakra-ui/react';
+
+import PaginationButton from './PaginationButton';
+import { ChevronLeftSmallIcon, ChevronRightSmallIcon } from '../icons';
+import Box from '../box';
+
+const Dots = '...';
+const range = (start: number, end: number): Array<number> => {
+  const length = end - start + 1;
+  return Array.from({ length }, (_, idx) => idx + start);
+};
+
+export interface Props {
+  count: number;
+  page: number;
+  siblingCount?: number;
+  onChange: (page: number) => void;
+}
+
+const Pagination: FC<PropsWithChildren<Props>> = (props) => {
+  const { onChange, count, page, siblingCount = 1 } = props;
+  const { paginationContainer, dots } = useMultiStyleConfig('Pagination');
+
+  const paginationRange = useMemo(() => {
+    // Default number of page buttons: firstPage + lastPage + currentPage + left side dots + right side dots
+    const pageButtonsCount = 5;
+    const totalPageButtonsCount = siblingCount + pageButtonsCount;
+
+    if (totalPageButtonsCount >= count) {
+      return range(1, count);
+    }
+
+    const leftSiblingIndex = Math.max(page - siblingCount, 1);
+    const rightSiblingIndex = Math.min(page + siblingCount, count);
+
+    const shouldShowLeftDots = leftSiblingIndex > 2;
+    const shouldShowRightDots = rightSiblingIndex < count - 2;
+
+    const firstPageIndex = 1;
+    const lastPageIndex = count;
+
+    if (!shouldShowLeftDots && shouldShowRightDots) {
+      const leftItemCount = 3 + 2 * siblingCount;
+      const leftRange = range(1, leftItemCount);
+
+      return [...leftRange, Dots, count];
+    }
+
+    if (shouldShowLeftDots && !shouldShowRightDots) {
+      const rightItemCount = 3 + 2 * siblingCount;
+      const rightRange = range(count - rightItemCount + 1, count);
+      return [firstPageIndex, Dots, ...rightRange];
+    }
+
+    if (shouldShowLeftDots && shouldShowRightDots) {
+      const middleRange = range(leftSiblingIndex, rightSiblingIndex);
+      return [firstPageIndex, Dots, ...middleRange, Dots, lastPageIndex];
+    }
+
+    return [];
+  }, [count, siblingCount, page]);
+
+  if (page === 0 || paginationRange.length < 2) {
+    return null;
```

Comment:
did you confirm that with a designer? For me this is a bit unexpected. I would expect a disabled pagination showing a single button if it only has 1 page

Reply:
Good point, will check that with designer

Reply:
Checked with designer, for now we will not show pagination component when there is only one page.

Reply:
Ok, but then you can remove the `currentPage === 0`, no?